components/earnings/commons/earnings_base.lua

```diff
@@ -0,0 +1,185 @@
+	end
+
+	local teamConditions = '([[participant::' .. team .. ']] OR [[extradata_participantteam::' .. team .. ']])'
+
+	return Earnings._calculate(teamConditions, args.year, args.mode, args.startYear, Earnings._divisionFactorOne)
+end
+
+---
+-- Calculates earnings for this participant in a certain mode
+-- @participantCondition - the condition to find the player/team
+-- @year - (optional) the year to calculate earnings for
+-- @mode - (optional) the mode to calculate earnings for
+function Earnings._calculate(conditions, year, mode, startYear, divisionFactor)
+	conditions = Earnings._buildConditions(conditions, year, mode, startYear)
+
+	if String.isNotEmpty(startYear) then
+		return Earnings._calculatePerYear(conditions, divisionFactor)
+	end
+
+	local lpdbQueryData = mw.ext.LiquipediaDB.lpdb('placement', {
+		conditions = conditions,
+		query = 'sum::prizemoney, mode',
+		groupby = 'mode asc'
+	})
+
+	local totalEarnings = 0
+
+	for _, item in ipairs(lpdbQueryData) do
+		if item['sum_prizemoney'] ~= nil then
+			local prizeMoney = item['sum_prizemoney']
+			totalEarnings = totalEarnings + (prizeMoney / divisionFactor(item['mode']))
+		end
+	end
+
+	return MathUtils._round(totalEarnings, 2)
+end
+
+function Earnings._calculatePerYear(conditions, divisionFactor)
+	local totalEarningsOfYear = {}
+	local totalEarnings = {}
+	totalEarnings.total = 0
+
+	local offset = 0
+	local count = 5000
+	while count == 5000 do
+		local lpdbQueryData = mw.ext.LiquipediaDB.lpdb('placement', {
+			conditions = conditions,
+			query = 'prizemoney, mode, date',
+			limit = 5000,
+			offset = offset
+		})
+		for _, item in pairs(lpdbQueryData) do
+			local prizeMoney = tonumber(item.prizemoney) or 0
+			local year = string.sub(item.date, 1, 4)
+			prizeMoney = prizeMoney / divisionFactor(item['mode'])
+			totalEarningsOfYear[year] = (totalEarningsOfYear[year] or 0) + prizeMoney
+		end
+		count = #lpdbQueryData
+		offset = offset + 5000
+	end
+
+	for year, earningsOfYear in pairs(totalEarningsOfYear) do
+		totalEarnings[tonumber(year)] = MathUtils._round(earningsOfYear, 2)
+		totalEarnings.total = totalEarnings.total + earningsOfYear
+	end
+	totalEarnings.total = MathUtils._round(totalEarnings.total, 2)
+
+	return totalEarnings
+end
+
+function Earnings._buildConditions(conditions, year, mode, startYear)
+	conditions = '[[date::!' .. _DEFAULT_DATE .. ']] AND [[prizemoney::>0]] AND ' .. conditions
+	if String.isNotEmpty(startYear) then
+		conditions = conditions .. ' AND (' ..
+			'[[date::>' .. startYear .. _FIRST_DAY_OF_YEAR .. ']] ' ..
+			'OR [[date::' .. startYear .. _FIRST_DAY_OF_YEAR .. ']])'
+	elseif String.isNotEmpty(year) then
+		conditions = conditions .. ' AND (' ..
+			'[[date::>' .. year .. _FIRST_DAY_OF_YEAR .. ']] ' ..
+			'OR [[date::' .. year .. _FIRST_DAY_OF_YEAR .. ']]' ..
+			') AND (' ..
+			'[[date::<' .. year .. _LAST_DAY_OF_YEAR .. ']] ' ..
+			'OR [[date::' .. year .. _LAST_DAY_OF_YEAR .. ']])'
+	end
```

Comment:
Can we use the YEAR modifier in condition string (`date_year`)? If it's not available, we should probably request it to be available on all date fields instead of specific ones.